src/HTTP.cxx

```diff
@@ -96,6 +96,11 @@ namespace influxdb::transports
         session.SetAuth(cpr::Authentication{user, pass, cpr::AuthMode::BASIC});
     }
 
+    void HTTP::setAuthToken(const std::string& token)
+    {
+        session.SetHeader(cpr::Header{{"Authorization", "Token " + token}});
```

Comment:
The problem is that the header will get overwritten later on in `HTTP::send` on line 107 in the same file.

Reply:
Here's a proposed solution:

https://github.dev/kasparthommen/influxdb-cxx/tree/fix-auth-token-overwriting

I have tested the code and it works.